driver-async/src/test/unit/com/mongodb/async/client/MongoDatabaseSpecification.groovy

```diff
@@ -349,6 +351,111 @@ class MongoDatabaseSpecification extends Specification {
         thrown(IllegalArgumentException)
     }
 
+    def 'should create ChangeStreamIterable correctly'() {
```

Comment:
These changestream tests were missing from the async database suite - so ported them across. Can split into a separate commit.